CMakeLists.txt

```
@@ -92,13 +92,23 @@ add_custom_command(
 file(RELATIVE_PATH DOXYGEN_BUILD_DIR_RELATIVE "${SOURCE_DOCS_DIR}" "${DOXYGEN_BUILD_DIR_CMAKE}")
 message(STATUS "DOXYGEN_BUILD_DIR_RELATIVE: ${DOXYGEN_BUILD_DIR_RELATIVE}")
 
+if(CMAKE_HOST_WIN32)
+    # On Windows, we have to build the documentation using only a single thread to
+    # avoid the build mysteriously taking forever.
+    # See https://github.com/LizardByte/Sunshine/pull/3361#issuecomment-2453549801.
+    set(DOXYGEN_NUM_THREADS 1)
+else()
+    set(DOXYGEN_NUM_THREADS 0)
+endif()
+
 # build docs
 add_custom_target(docs ALL
         COMMENT "Building Doxygen documentation"
         WORKING_DIRECTORY "${SOURCE_DOCS_DIR}"
         COMMAND ${CMAKE_COMMAND} -E env
             READTHEDOCS_OUTPUT=${DOXYGEN_BUILD_DIR_RELATIVE}
             READTHEDOCS_VERSION=${DOXYGEN_PROJECT_VERSION}
+            READTHEDOCS_THREADS=${DOXYGEN_NUM_THREADS}
```

```suggestion
            DOXYCONFIG_THREADS=${DOXYGEN_NUM_THREADS}
```

The other two env variables are standard readthedocs variables: https://docs.readthedocs.io/en/stable/reference/environment-variables.html